Game/space-invaders/src/Game.cpp

```diff
@@ -27,30 +26,91 @@ void Game::Init()
     m_SpriteRenderer = std::make_unique<SpriteRenderer>(spriteShader);
     
     // Create Player
-    m_PlayerManager = std::make_unique<PlayerManager>();
+    m_PlayerManager = std::make_unique<PlayerManager>(*this);
     m_PlayerManager->CreatePlayer(WIDTH, HEIGHT);
 }
 
 void Game::Update(float deltaTime)
 {
-    // std::cout << "Game running\n";
+    // move player projectiles
+    for (auto& projectile : m_PlayerProjectiles)
+    {
+        // check boundaries
+        if (projectile.Position.y <= 0)
+        {
+            projectile.Destroyed = true;
+            continue;
+        }
+        
+        projectile.Position.y -= projectile.Velocity.y * deltaTime;
+    }
 }
 
 void Game::ProcessInput(float deltaTime, const Input& input)
 {
-    // std::cout << "Getting input\n";
     m_PlayerManager->ProcessInput(deltaTime, input, WIDTH);
 }
 
 void Game::Render()
 {
-    // Render Player
     m_PlayerManager->Render(*m_SpriteRenderer);
+
+    RenderProjectiles();
+    RemoveDestroyedProjectiles();
 }
 
 void Game::Close()
 {
-    std::cout << "Closing game...\n";
-
     ResourceManager::ClearAll();
 }
+
+void Game::RenderProjectiles() const
+{
+    for (auto& projectile : m_PlayerProjectiles)
+    {
+        projectile.Draw(*m_SpriteRenderer);
+    }
+}
+
+void Game::AddEnemyProjectile(GameObject&& projectile)
+{
+    GameObject newProjectile = projectile;
```

Comment:
No need to set to a local var and them emplace_back, just do m_Projectiles.emplace_back(std::move(projectile)), for enemy and player